### PR TITLE
fix reporting of JSON export failure

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Test Plan
On Windows...
- Check that can export entities to directories other than C:/ root directory.
- Check that a message box pops up indicating failure when try to export file to C:/ root directory.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 657f8fdad56d5241c0d24c1219192358fae28cf1 available at: https://gist.github.com/2e84d0c2ad81d1b89f4fdc3adfd80457<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8075.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8075.dmg

**Comment by Nex-Pro** [comment]
**PASS**

@ctrlaltdavid Seems fine.

File: libraries/ui/src/FileDialogHelper.cpp [hunk 1]
```diff
@@ -95,6 +95,8 @@ bool FileDialogHelper::urlIsWritable(const QUrl& url) {
 
     // No file, get the parent directory and check if writable
     return QFileInfo(fileInfo.absoluteDir().absolutePath()).isWritable();
+    // Note: Does not correctly detect on Windows that the OS drive root (i.e., C:/) cannot be written to. This even if turn on
```
Comment (jherico) [inline libraries/ui/src/FileDialogHelper.cpp]: The permissions for the OS drive root are more complicated than just 'can I write here'.  On windows 10, by default, you can create folders and modify files without special permissions.  Only creating new files (not folders) requires administrative rights.  This granularity of rights isn't exposed by the Qt framework.  

One solution would be to special case drive roots in a Q_OS_WIN block.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 775656eb1d8e5daaabc8e0861738c7ee06d658bc available at: https://gist.github.com/2e84d0c2ad81d1b89f4fdc3adfd80457<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8075.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8075.dmg

**Pull request merged**